engine/src/main/java/com/datatorrent/stram/engine/Node.java

```diff
@@ -485,6 +485,10 @@ protected void deactivateSinks()
 
   void checkpoint(long windowId)
   {
+    if (operator instanceof Operator.CheckpointNotificationListener) {
```

Comment:
I think this call should be made irrespective of whether the operator is stateless or not. Here is the case I am thinking of, If I want to serialize the operator state in my own way I should be able to do in call. I can mark all my properties transient and use this call to serialize my state. I can do using storage agent too but this could be easier and faster way.. 

Just a thought